### PR TITLE
Fix the compilation chunks order bug

### PR DESCRIPTION
**Pull request by SGospodinov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -256,22 +251,14 @@ class StaticComponentEnveloper {
...
-      })
-      assets.css = assets.css.concat(css)
-    })
-
-    // Duplicate css assets can occur on occasion if more than one chunk
-    // requires the same css.
-    assets.css = new Set(assets.css)
-
-    return assets
+      const fileTypeFilter = filterRegEx =>
```
Comment (EdwardLuke7) [inline index.js]: Much nicer to read this section now :)

File: index.js [hunk 2]
```diff
@@ -237,14 +237,9 @@ class StaticComponentEnveloper {
...
-    const assets = {
-      js: [],
-      css: []
-    }
 
-    chunks.forEach((chunk) => {
-      // Prepend the public path to all chunk files
-      const chunkFiles = [].concat(chunk.files).map((chunkFile) => {
+    return chunks.reduce((assets, chunk) => {
+      const chunkFiles = chunk.files.map((chunkFile) => {
```
Comment (EdwardLuke7) [inline index.js]: I wonder if the [].concat was there to handle the case where chunks.files was undefined (or some such shenanigans?). Could you find any docs to give us some certainty around what we expect `chunks.files` to be - particularly in the case that there are indeed no chunks generated?

Comment (SGospodinov) [reply index.js]: That's a good point. The best thing that I found is this https://webpack.js.org/api/stats/#structure document, according to which the chunks are a `list`. However, they do not specify if the list could be `null` or `undefined`, so I'll try to guarantee that we are not trying to iterate over an `undefined`.

File: index.js [hunk 3]
```diff
@@ -239,7 +239,7 @@ class StaticComponentEnveloper {
     const publicPath = this.getPublicPath({ compilation })
 
     return chunks.reduce((assets, chunk) => {
-      const chunkFiles = chunk.files.map((chunkFile) => {
+      const chunkFiles = [...chunk.files].map((chunkFile) => {
```
Comment (EdwardLuke7) [inline index.js]: I think you'll get a `TypeError` using the spread operator on a non-iterable (like null or undefined). But maybe it's not worth overthinking it! This way at least we make sure we're operating on a plain ole' JS array. Yay for no docs.

Comment (SGospodinov) [reply index.js]: You are absolutely right about the TypeError. I'll revers it to the array concat one.

**Pull request merged**